### PR TITLE
ICU-21249 BRS68RC: Fixes and changes for the ICU4C Samples.

### PR DESCRIPTION
**Pull request by jefgen**
This PR makes changes/fixes for the ICU 68 BRS task for "ICU4C Samples", in order to get the samples to compile with some of the changes made in ICU 68. (The change to use C++/C99 `true`/`false` instead of of the custom ICU macros for them).

Changes in this PR include:
- Change to use `true`/`false` instead of the custom `TRUE`/`FALSE` macros in order to fix compilation errors.
- Adding casts to prevent/silence warnings.
- Some clean-up to the "ufortune" and "ugrep" Visual Studio project `.vcxproj` files.
- Setting the codepage to UTF-8 in `samplecheck.bat` in order to prevent garbled output when running the samples.

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21249
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/samples/coll/coll.cpp [hunk 1]
```diff
@@ -244,12 +244,12 @@ UBool processCollator()
  */
 int main(int argc, const char** argv) 
 {
-    if (processOptions(argc, argv, opts) != TRUE || opt_help) {
+    if (static_cast<bool>(processOptions(argc, argv, opts)) != true || static_cast<bool>(opt_help)) {
```
Comment (daniel-ju) [inline icu4c/source/samples/coll/coll.cpp:247]: Kind of a nit, but do we still need to compare equal/not equal to true here, versus using/not using `!` (same in other places/files)?

Comment (jefgen) [reply icu4c/source/samples/coll/coll.cpp:247]: Oh good point, since these are `bool`s now, we should be able to just use an if statement. :) 
I'll force-push an update later.

**Review by daniel-ju — APPROVED** [review]

File: icu4c/source/samples/ucnv/convsamp.cpp [hunk 2]
```diff
@@ -342,7 +342,7 @@ UErrorCode convsample_05()
         
         ucnv_toUnicode(conv, &target, targetLimit, 
                        &source, sourceLimit, NULL,
-                       feof(f)?TRUE:FALSE,         /* pass 'flush' when eof */
+                       feof(f)?true:false,         /* pass 'flush' when eof */
```
Comment (erik0686) [inline icu4c/source/samples/ucnv/convsamp.cpp:345]: Could this be the same as Daniel mentioned before? removing the condition? Same for cases below?

Comment (jefgen) [reply icu4c/source/samples/ucnv/convsamp.cpp:345]: Hmm, good question...

The documentation for the function `feof` only says that it returns a "nonzero value":
> The feof function returns a nonzero value if a read operation has attempted to read past the end of the file; it returns 0 otherwise.

However, the function `ucnv_toUnicode` takes a `UBool`, which is a `typedef` for `int8_t`. This means that it **only** wants a value of `1` for things that are "true". So because of this I think we need to keep the condition here, as we need to possibly handle cases where `feof` returns back something like 2.
In other words, we need it to work like this: ` feof returns 2 --> true --> 1`.
So I think this needs to stay as-is.

Comment (erik0686) [reply icu4c/source/samples/ucnv/convsamp.cpp:345]: Ah that is a good point :D Thanks!

**Review by erik0686 — APPROVED** [review]
